components/Tabs2/src/MDCTabBarView.m

```diff
@@ -16,4 +16,51 @@
 
 @implementation MDCTabBarView
 
+#pragma mark - Properties
+
+- (void)setItems:(NSArray<UITabBarItem *> *)items {
+  NSAssert([NSThread isMainThread], @"Item array may only be set on the main thread");
+  NSParameterAssert(items != nil);
+
+  if (_items != items && ![_items isEqual:items]) {
+    _items = [items copy];
+
+    // Determine new selected item, defaulting to the first item.
+    UITabBarItem *newSelectedItem = _items.firstObject;
```

Comment:
I think since the selected item can be `nil`, it's better to default to that.  What do you think?

Reply:
I thought this is a little bit tricky. The original behavior of MDCTabBar will select the first item from the items, if the default is nil, which child view controller should be rendered? Or you believe this can be handled by another init method? 
Thanks for rising this though!

Reply:
UIKit does change the selectedItem to nil when the selected item no longer exists in the new items array:

```objc
UITabBar *tabBar = [[UITabBar alloc] initWithFrame:CGRectZero];
UITabBarItem * item1 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:1];
UITabBarItem * item2 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:2];
UITabBarItem * item3 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:3];
tabBar.items = @[item1, item2 ];
tabBar.selectedItem = item1;
NSLog(@"%@", @(tabBar.selectedItem.tag)); // 1
tabBar.items = @[item1 ];
NSLog(@"%@", @(tabBar.selectedItem.tag)); // 0
```

I recommend following the UIKit behavior and writing a test that validates the UIKit behavior over time in case it changes from one iOS version to another.

Reply:
Sure! I changed it to the same behavior with the UIKit.
I try to understand your meaning of test against But what do you exactly mean when you mention the test against the UIKit?

Reply:
Something like so, just for consistency validation in terms of expected behavior:

```objc
- (void)testUITabBarSelectedItemBecomesNilWhenRemovedFromItems {
  // Given
  UITabBar *tabBar = [[UITabBar alloc] initWithFrame:CGRectZero];
  UITabBarItem * item1 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:1];
  UITabBarItem * item2 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:2];
  UITabBarItem * item3 = [[UITabBarItem alloc] initWithTitle:@"Hi" image:nil tag:3];
  tabBar.items = @[item1, item2 ];
  tabBar.selectedItem = item1;

  // When
  tabBar.items = @[item1 ];

  // Then
  XCTAssertNil(tabBar.selectedItem);
```

Reply:
+1 to following UIKit behavior (don't automatically pick a selection). 

Regarding the test, I don't know that it's worth writing a unit test to monitor UIKit's behavior. Even if UITabBar does change in the future, there are only three outcomes that I can think of: rushing to change our API to match (without necessarily deciding it's the best option); turning off the test; or turn off the test, file an Issue to investigate changing the behavior, and then possibly change it.

Without the test, we would have discovered the difference once a client asked us to match UIKit's behavior.  Until a client is dissatisfied with the behavior at that point, I don't think it's worth the canary.

Just my thoughts, and @jverkoey trumps as the TL if he really wants the test.

Reply:
Canaries are helpful when it comes time to validating our behavior against the expectations of new iOS versions, but I don't feel particularly strongly about adding it. The risk of UIKit changing this behavior to something that would be surprising for clients is pretty low.

Reply:
I mean:
/// Tab bars should by default select nil in their items array. The behavior should also consistent
/// with the UIKit
- (void)testSelectsNilByDefault {
  // Given
  UITabBar *tabBar = [[UITabBar alloc] initWithFrame:CGRectZero];
  tabBar.items = @[ _itemA, _itemB, _itemC ];
  _tabBarView.items = @[ _itemA, _itemB, _itemC ];

  // Then
  XCTAssertNil(_tabBarView.selectedItem);
  XCTAssertNil(tabBar.selectedItem);
}

Does this looks good to you?

Reply:
So shall I left with no tests in this case?

Reply:
Thanks for everyone's input! So I just leave with no tests against UIKit for now...

Reply:
To give more clear guidance here: I think we'll be ok if we don't write the UIKit canary so it's ok not to add such a test in this PR.